File: remove-duplicates-from-sorted-array.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if len(nums) <= 0:
            return 0
        
        prev = nums[0]
        prev_idx = 0
        for idx, num in enumerate(nums[1:]):
            if num != prev:
                prev_idx += 1
                nums[prev_idx] = num
                prev = num
                
        return prev_idx + 1

File: remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        if(nums.length <= 0) {
            return 0;
        }
        int prev = nums[0];
        int prev_idx = 0;
        for(int i = 1; i < nums.length; i++) {
            if(prev != nums[i]) {
                nums[++prev_idx] = nums[i];
                prev = nums[i];
            }
        }
        return prev_idx + 1;
    }
}